tests/ui/doc/doc.stderr

```diff
@@ -1,190 +1,166 @@
-error: you should put `foo_bar` between ticks in the documentation
-  --> $DIR/doc.rs:8:9
+error: item in documentation is missing backticks
+  --> $DIR/doc.rs:9:9
    |
 LL | /// The foo_bar function does _nothing_. See also foo::bar. (note the dot there)
-   |         ^^^^^^^
+   |         ^^^^^^^ help: try: ``foo_bar``
    |
    = note: `-D clippy::doc-markdown` implied by `-D warnings`
 
-error: you should put `foo::bar` between ticks in the documentation
-  --> $DIR/doc.rs:8:51
+error: item in documentation is missing backticks
+  --> $DIR/doc.rs:9:51
    |
 LL | /// The foo_bar function does _nothing_. See also foo::bar. (note the dot there)
-   |                                                   ^^^^^^^^
+   |                                                   ^^^^^^^^ help: try: ``foo::bar``
 
-error: you should put `Foo::some_fun` between ticks in the documentation
-  --> $DIR/doc.rs:9:83
+error: item in documentation is missing backticks
+  --> $DIR/doc.rs:10:83
    |
 LL | /// Markdown is _weird_. I mean _really weird_. This /_ is ok. So is `_`. But not Foo::some_fun
-   |                                                                                   ^^^^^^^^^^^^^
+   |                                                                                   ^^^^^^^^^^^^^ help: try: ``Foo::some_fun``
 
-error: you should put `a::global:path` between ticks in the documentation
-  --> $DIR/doc.rs:11:15
+error: item in documentation is missing backticks
+  --> $DIR/doc.rs:12:15
    |
 LL | /// Here be ::a::global:path.
-   |               ^^^^^^^^^^^^^^
+   |               ^^^^^^^^^^^^^^ help: try: ``a::global:path``
```

Comment:
Side note: This lint case could be improved by including the first double colon for global identifiers. I've created a new issue #7914 to keep this in mind. This is not a result of your changes. Therefore, I would suggest not to fix it in this PR, also to keep the changes manageable :upside_down_face: